src/inmanta/parser/plyInmantaLex.py

```diff
@@ -246,3 +221,40 @@ def t_ANY_error(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
 
 # Build the lexer
 lexer = lex.lex()
+
+
+def safe_decode(token: lex.LexToken, warning_message: str, start: int = 1, end: int = -1) -> str:
+    """
+    Check for the presence of an invalid escape sequence (e.g. "\.") in the value attribute of a given token.  # noqa: W605
+    This function assumes to be called from within a t_STRING or a t_MLS rule.
+
+    - Python < 3.12 raises a DeprecationWarning when encountering an invalid escape sequence
+    - Python 3.12 will raise a SyntaxWarning
+    - Future versions will eventually raise a SyntaxError
+    (see https://docs.python.org/3.12/whatsnew/3.12.html#other-language-changes )
+
+    :param token: The token whose value we want to decode.
+    :param warning_message: The warning message to display.
+    :param start: Start of the value slice (To only decode the characters after the leading quotation mark(s))
+    :param end: End of the value slice (To only decode the characters before the trailing quotation mark(s))
+
+    :return: The token value as a python str.
+    """
+
+    try:
+        # This first block will try to decode the value and turn any deprecation warning into an actual Exception.
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error", message="invalid escape sequence", category=DeprecationWarning)
+            value: str = bytes(typing.cast(str, token.value)[start:end], "utf_8").decode("unicode_escape")
+    except DeprecationWarning:
+        # If the first block did actually encounter an invalid escape sequence, we have to decode the value again, this time
+        # ignoring the python warning, and raising a warning of our own.
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="invalid escape sequence", category=DeprecationWarning)
```

Comment:
I think you actually got this right before: non-deprecation warnings would have already been emitted (we didn't catch them) so we can safely ignore any warning here.
```suggestion
        # ignoring this or any other python warning that has already been emitted, and raising a warning of our own.
        with warnings.catch_warnings():
            warnings.filterwarnings("ignore")
```

Reply:
Ah yes indeed :+1: